src/folding/nova/ivc.rs

```diff
@@ -0,0 +1,272 @@
+use ark_crypto_primitives::sponge::{poseidon::PoseidonConfig, Absorb};
+use ark_ec::{CurveGroup, Group};
+use ark_ff::PrimeField;
+use ark_relations::r1cs::ConstraintSynthesizer;
+use ark_relations::r1cs::ConstraintSystem;
+use ark_std::rand::Rng;
+use ark_std::{One, Zero};
+use core::marker::PhantomData;
+
+use super::circuits::{AugmentedFCircuit, FCircuit};
+use super::{
+    nifs::NIFS,
+    traits::{NovaR1CS, NovaTranscript},
+    CommittedInstance, Witness,
+};
+use crate::ccs::r1cs::R1CS;
+use crate::frontend::arkworks::{extract_r1cs, extract_z}; // TODO once Frontend trait is ready, use that
+use crate::pedersen::{Params as PedersenParams, Pedersen};
+use crate::transcript::Transcript;
+use crate::{unwrap_or_return_err, Error};
+
+pub struct IVC<C1, C2, FC, Tr>
```

Comment:
Maybe add some minimal docs on what is that intended to do.

Reply:
Agree, updated.